FEXCore/Source/Interface/IR/Passes/ConstProp.cpp

```diff
@@ -542,7 +543,8 @@ bool ConstProp::ConstantPropagation(IREmitter *IREmit, const IRListView& Current
       auto Op = IROp->CW<IR::IROp_StoreMem>();
       auto AddressHeader = IREmit->GetOpHeader(Op->Addr);
 
-      if (AddressHeader->Op == OP_ADD && AddressHeader->Size == 8) {
+      if (AddressHeader->Op == OP_ADD &&
+          (AddressHeader->Size == 8 || AddressHeader->Size == 4)) {
         auto [OffsetType, OffsetScale, Arg0, Arg1] = MemExtendedAddressing(IREmit, IROp->Size, AddressHeader);
```

Comment:
Probably need some additional work here to ensure that the `MemExtendedAddressing` for the FPRClass Load/StoreMem fits within ARM's vector loadstore limits.
x86 SIB addressing is quite a bit more flexible so it can easily not fit within ARM constraints.
Might need a version of the helper function for GPR loadstores and Vector loadstores.

Reply:
Thanks - still getting to know the ISAs. Will spend some time today trying to understand the difference between arm and intel addressing modes.